openmc/model/surface_composite.py

```diff
@@ -773,16 +763,77 @@ def _make_ccw(self, points):
         -------
         ordered_points : the input points ordered counter-clockwise
         """
+        points = np.asarray(points, dtype=float)
+        check_iterable_type('points', points, float, min_depth=2, max_depth=2)
+        check_length('points', points[0, :], 2, 2)
+
+        # If the last point is the same as the first, remove it and make sure
+        # there are still at least 3 points for a valid polygon.
+        if np.allclose(points[0, :], points[-1, :]):
+            points = points[:-1, :]
+        check_length('points', points, 3)
+
+        # Check if polygon is self-intersecting by comparing edges pairwise
+        n = len(points)
+        is_self_intersecting = False
+        for i in range(n):
+            p1x, p1y = points[i, :]
+            p2x, p2y = points[(i + 1) % n, :]
+            for j in range(i, n):
+                p3x, p3y = points[j, :]
+                p4x, p4y = points[(j + 1) % n, :]
+
```

Comment:
@eepeterson is it possible that you forgot to commit something here?  
Seems like there should be a check that could potentially set `is_self_intersecting` to `False`, no?

Reply:
yeah I haven't found an easy way of checking if a polygon doesn't self intersect without writing a lot of code or pulling in another dependency so I stalled on this just a bit. To catch edge cases and such it can be a bit involved.